### PR TITLE
TRITON-2378 Update sshpk for ed25519 support

### PR DESCRIPTION
**Pull request by travispaul**
## Testing notes

- Confirmed that `make check test` succeeded.
- Pulled these changes into [smartdc-auth](https://github.com/TritonDataCenter/node-smartdc-auth/compare/master...TRITON-2378) and confirmed that  `make check test` succeeded
- Pulled these changes into [cloudapi](https://github.com/TritonDataCenter/sdc-cloudapi/compare/master...TRITON-2378) and confirmed that  `make check` succeeded and that I could install the changes via `sdcadm update -C experimental cloudapi@TRITON-2378-20231113T181023Z-g073fbb7`
- Finally, I pulled in these change to [node-triton](https://github.com/TritonDataCenter/node-triton/compare/TRITON-2378) (via smartdc-auth), removed all keys from my account except for an ED25519 key and confirmed that I could make Cloud API requests and provision instances.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,20 +1,20 @@
 {
   "name": "http-signature",
   "description": "Reference implementation of Joyent's HTTP Signature scheme.",
-  "version": "1.3.6",
+  "version": "1.4.0",
   "license": "MIT",
-  "author": "Joyent, Inc",
+  "author": "MNX Cloud (mnx.io)",
   "contributors": [
     "Mark Cavage <mcavage@gmail.com>",
```
Comment (danmcd) [inline package.json:8]: Add yourself here?

**Review by danmcd — APPROVED** [review]
Optional nit, fix it or not at your discretion.

PLEASE make sure the testing notes you have here also land in the TRITON-2378 ticket itself.

File: package.json (hunk 1 above)
Comment (travispaul) [reply package.json:8]: I'd prefer to omit myself from that list since it's such a small change.

**Comment by danmcd** [comment]
Just add the testing notes to the ticket then.

**Pull request merged**